stage2/client.py

```diff
@@ -146,20 +146,56 @@ def udp_receive_messages(self, udp_socket):
                 print(
                     f"Key error: {e}. The received message does not have the expected format."
                 )
+                
+    def prompt_and_validate_username(self) -> str:
+        max_bytes_in_username = 255
+        
+        while True:
+            username = input("Enter your username: ")
+            
+            if len(username.encode()) > max_bytes_in_username:
+                print("The username exceeds the maximum character limit.")
+                
+            elif len(username) <= 0:
+                print("Username is required.")
+                
+            else:
+                return username
+    
+    def get_valid_menu_selection(self) -> str:
+        while True:
+            operation_code = input("Create (1) or Join (2) a chatroom? ")
+            
+            if operation_code == "1" or operation_code == "2":
+                return operation_code
+            
+            else:
+                print("Please enter 1 or 2.")
+    
+    def get_and_validate_room_name(self) -> str:
+        max_bytes_in_room_name = 256
```

Comment:
仕様ではルーム名の最大バイト数は2^8バイトとありますが、
ヘッダを作成する時、RoomNameSizeを1バイトにしてます。
256を表すには2バイト必要になってしまうので、ルーム名も255バイトまでにした方が良いと思います。
自分の理解が間違ってたら指摘して下さい💦
```suggestion
        max_bytes_in_room_name = 255
```

Reply:
僕も255でいいかなって思ってます